Bottom: 298fd670441cc02b98cc7fab38b55254032cf4c6
Top:    ab31537b275d4dd0c6a20822b458338f6f5b6bd2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 23:23:51 +0000

Fix a few random places where we fail badly if the board outline polygon computes to NULL

Usually, the board outline polygon should default to being the world region, but if a
polygon computation fails, it may return NULL, and we can crash in a few places if that
happens.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 3d1b6cc..a27a6b3 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1692,6 +1692,8 @@ fill_board_outline_holes (hidGC gc, const BoxType *drawn_area)
   struct outline_info info;
 
   ensure_board_outline ();
+  if (PCB->Data->outline == NULL)
+    return;
 
   memset (&polygon, 0, sizeof (polygon));
   polygon.Clipped = PCB->Data->outline;
